Stats: Add mobile promo card to Ads page.

a8ck3n opened the pull request:
#### Proposed Changes

Ads the mobile promo card to the bottom of the Ads page.

#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Open the Calypso Live link.
* Navigate to Stats → Ads.
* Scroll to the bottom and confirm the card is visible.

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #72675.

a8ck3n committed:
Stats: Add moble promo card to Ads page.

a8ck3n committed:
Minor tweak to props deconstruction.

dognose24 commented on client/my-sites/stats/wordads/index.jsx:
```diff
@@ -246,6 +257,13 @@ class WordAds extends Component {
 							</div>
 
+							<PromoCards
+								isJetpack={ isJetpack }
```
The prop `isJetpack` [seems not to be necessary anymore](https://github.com/Automattic/wp-calypso/pull/71286/files#diff-d0adee9cff76fc082d6cdc6ebfa2dde684cfa28d52007d8c69599eb26b943f87L14).

dognose24 reviewed (approved):
Works as described! I found a deprecated prop of `PromoCards`, which can be removed.

jsnmoon commented on client/my-sites/stats/promo-cards/index.jsx:
```diff
@@ -72,7 +75,10 @@ export default function PromoCards( { isOdysseyStats, slug, pageSlug } ) {
 		// This will allow us to account for integration across different pages.
 		// FORMAT: calypso_stats_PAGE_mobile_cta_EVENT
-		const tracksEventName = `calypso_stats_traffic_mobile_cta_${ event.replaceAll( '-', '_' ) }`;
+		const tracksEventName = `calypso_stats_${ pageSlug }_mobile_cta_${ event }`.replaceAll(
+			'-',
+			'_'
+		);
```
Could we add a code comment here listing all the different permutations of what `tracksEventName` can be?

> "Don’t programmatically create event names. They should be hardcoded so they are easily searchable in the code base."
- From "General Naming Conventions" on PCYsg-4S2-p2.

a8ck3n replied:
I tried to set up the event names based on those guidelines. Part of the initial design was to have unique events for each page the CTA is shown on which is why we build the event based on the provided pageSlug prop.

There are 6 possible click events per page. I've added a comment listing those out (currently 18 possible events) but it will fall out of date if the PromoCards component is used on any other pages.

I'm not sure I have a better solution than programmatically generating the event names. If you have any suggestions, please let me know.

jsnmoon reviewed (approved):
The code looks good and appropriate events fire on render & upon being clicked. 

This PR is good to merge after addressing my and Dognose's feedback.

a8ck3n committed:
Remove unused isJetpack property.

a8ck3n committed:
Clean up PromoCard call sites.

Unrelated cleanup of PromoCard call sites to remove the no longer needed isJetpack property.

a8ck3n committed:
Add list of known click events.

These are supposed to be unique to the call site so they are generated based on the provided slug. For this reason, the comment will fall out of date if not updated for each new call site.
